### PR TITLE
isDataAvailableForKey should be public to easier to check data available

### PR DESCRIPTION
**Pull request by kennic**
… so you don't need to fetch all data to get a certain data.
Sometimes the data of a user is very long but we just need to get the "username" which is already fetched.
This PR also avoid crashing when we try to get data from key which is not fetched. It should returns nil.

So we could check or get data by either ways:
```
if user.isDataAvailable(forKey: "username") {
 // ok
}
```
or
```
if let username = user["username"] {
// username is available
}
else {
// username is unavailable, please fetch it
}
```

### CODE REVIEW

**Review by stephanboner — CHANGES_REQUESTED** [review]
Looks good to me in general but why did you include the change in ParseUI into this PR? Is this on purpose?

**Comment by kennic** [comment]
> Looks good to me in general but why did you include the change in ParseUI into this PR? Is this on purpose?

It's on purpose, it's just a minor fix to silent the "variable not used" warning

**Review by stephanboner — APPROVED** [review]
Well then it looks good to me

File: ParseUI/Classes/SignUpViewController/PFSignUpView.m [hunk 1]
```diff
@@ -213,7 +213,7 @@ - (void)layoutSubviews {
         frame.origin.y = currentY + loginButtonTopInset;
         _signUpButton.frame = frame;
 
-        currentY = CGRectGetMaxY(frame);
```
Comment (dielsonsales) [inline ParseUI/Classes/SignUpViewController/PFSignUpView.m:216]: Why is this being kept commented though?

**Comment by dplewis** [comment]
@mtrezza What do you want to do with this?

**Comment by mtrezza** [comment]
Closing a superseded by #1756

**Pull request closed without merging**